crates/starknet-os/src/hints/kzg.rs

```diff
@@ -0,0 +1,269 @@
+use std::collections::HashMap;
+use std::io::{self, Read};
+use std::path::Path;
+
+use c_kzg::{Blob, KzgCommitment, BYTES_PER_FIELD_ELEMENT};
+use cairo_vm::hint_processor::builtin_hint_processor::hint_utils::{
+    get_ptr_from_var_name, get_relocatable_from_var_name,
+};
+use cairo_vm::hint_processor::hint_processor_definition::HintReference;
+use cairo_vm::serde::deserialize_program::ApTracking;
+use cairo_vm::types::exec_scope::ExecutionScopes;
+use cairo_vm::types::relocatable::MaybeRelocatable;
+use cairo_vm::vm::errors::hint_errors::HintError;
+use cairo_vm::vm::vm_core::VirtualMachine;
+use cairo_vm::Felt252;
+use indoc::indoc;
+use num_bigint::BigInt;
+use num_traits::{Num, One, Zero};
+
+const FIELD_ELEMENTS_PER_BLOB: usize = 4096;
+const COMMITMENT_BYTES_LENGTH: usize = 48;
+const COMMITMENT_LOW_BIT_LENGTH: usize = (COMMITMENT_BYTES_LENGTH * 8) / 2;
+const BLOB_SUBGROUP_GENERATOR: &str = "39033254847818212395286706435128746857159659164139250548781411570340225835782";
+const BLS_PRIME: &str = "52435875175126190479447740508185965837690552500527637822603658699938581184513";
+
+fn fft(coeffs: &[BigInt], generator: &BigInt, prime: &BigInt, bit_reversed: bool) -> Result<Vec<BigInt>, io::Error> {
+    fn _fft(coeffs: &[BigInt], group: &[BigInt], prime: &BigInt) -> Vec<BigInt> {
+        if coeffs.len() == 1 {
+            return coeffs.to_vec();
+        }
+
+        let f_even = _fft(
+            &coeffs.iter().step_by(2).cloned().collect::<Vec<_>>(),
+            &group.iter().step_by(2).cloned().collect::<Vec<_>>(),
+            prime,
+        );
+        let f_odd = _fft(
+            &coeffs.iter().skip(1).step_by(2).cloned().collect::<Vec<_>>(),
+            &group.iter().step_by(2).cloned().collect::<Vec<_>>(),
+            prime,
+        );
+
+        let group_mul_f_odd: Vec<BigInt> =
+            group.iter().take(f_odd.len()).zip(f_odd.iter()).map(|(g, f)| (g * f) % prime).collect();
+
+        let mut result = Vec::with_capacity(coeffs.len());
+        for i in 0..f_even.len() {
+            result.push((f_even[i].clone() + &group_mul_f_odd[i]) % prime);
+        }
+        for i in 0..f_even.len() {
+            // Ensure non-negative diff
+            let diff = (f_even[i].clone() - &group_mul_f_odd[i] + prime) % prime;
+            result.push(diff);
+        }
+
+        result
+    }
+
+    if coeffs.is_empty() {
+        return Ok(vec![]);
+    }
+
+    let coeffs_len = coeffs.len();
+    assert!(coeffs_len.is_power_of_two(), "Length is not a power of two.");
+
+    let mut group = vec![BigInt::one()];
+    for _ in 1..coeffs_len {
```

Comment:
For similarity with the Python code.
```suggestion
    for _ in 0..(coeffs_len - 1) {
```